helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -249,23 +220,33 @@ private boolean validateOfflineInstancesLimit(final ResourceControllerDataProvid
     return true;
   }
 
+  private void updateWagedRebalancer(WagedRebalancer wagedRebalancer, ClusterConfig clusterConfig) {
```

Comment:
Should we check things on the WagedRebalancer too? is closed?

Reply:
That's a good point. For now, an explicit check would be optional. If the rebalancer has been closed, the computing will fail. One concern might be metric. It will be unregistered after the close.

Let me add a TODO to the close() method. I would prefer to keep this PR simple for now.

Reply:
One more comment, for now, the close is only called in 2 conditions:
1. the instance is GCed. So there should be no one calling any methods.
2. The controller is shutdown. There won't be a pipeline running after. So there should be no one calling any methods.

That's why I don't think we need to do this check immediately.

Reply:
True, but adding a check is safer and we could add a TODO to remove with better design?

Reply:
I'd prefer adding a TODO to add this check properly. Since that is not one or two lines of change, let's not squeeze it into this PR.

Reply:
Will we be tracking this work with an issue?